pkg/proxy/iptables/proxier.go

```diff
@@ -1225,7 +1225,21 @@ func (proxier *Proxier) syncProxyRules() {
 		// metrics come out right, so we just compute them and throw them away.
 		if tryPartialSync {
 			svcNN := svcName.NamespacedName.String()
-			if !serviceChanged.Has(svcNN) && !endpointsChanged.Has(svcNN) {
+			canSkip := !serviceChanged.Has(svcNN) && !endpointsChanged.Has(svcNN)
+			if canSkip && len(existingNATChains) > 0 {
+				// FIXME: This shouldn't happen, but apparently it does
+				// sometimes. https://issues.k8s.io/121362.
+				for _, ep := range allLocallyReachableEndpoints {
+					if epInfo, ok := ep.(*endpointInfo); ok {
+						if _, exists := existingNATChains[epInfo.ChainName]; !exists {
+							klog.InfoS("Service has not changed but endpoint chain is missing?", "service", svcNN, "chain", epInfo.ChainName)
```

Comment:
we want people hitting this to report it, we have things like this
```
    logger.V(4).Info("Synchronizing unbound PersistentVolumeClaim, volume already bound to different claim by controller, THIS SHOULD NEVER HAPPEN", "PVC", klog.KObj(claim), "boundClaim", klog.KRef(volume.Spec.ClaimRef.Namespace, volume.Spec.ClaimRef.Name))
```
I think we can append the [IT SHOULD NOT HAPPEN] or similar so we get notified of this problem


A bit of stretch, but can this count as an `IptablesPartialRestoreFailuresTotal`?

Reply:
I didn't read this well , I think we better hold on instrumenting the entire kube-proxy to have a better reproduction and confirmation kube-proxy is the root cause https://github.com/kubernetes/kubernetes/pull/122048/commits/ce66573d9817ee6c189fe97e3ec623bb44b49c75#r1409112716